src/Raven.Server/ServerWide/ShardingStore.cs

```diff
@@ -86,6 +90,81 @@ public bool HasActiveMigrations(string database)
             return false;
         }
 
+        public void ClearPublishedUrls()
+        {
+            using (_serverStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext ctx))
+            using (var tx = ctx.OpenWriteTransaction())
+            {
+                PublishedUrls.Clear(ctx);
+                tx.Commit();
+            }
+        }
+
+        public string ReadPublishedUrls()
+        {
+            using (_serverStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext ctx))
+            using (ctx.OpenReadTransaction())
+            {
+                var val = PublishedUrls.Read(ctx);
+                return ctx.ReadObject(val.ToJson(), "read-published-urls").ToString();
+            }
+        }
+
+        public DocumentConventions DocumentConventionsForShard =>
+            new()
+            {
+                SendApplicationIdentifier = DocumentConventions.DefaultForServer.SendApplicationIdentifier,
+                MaxContextSizeToKeep = DocumentConventions.DefaultForServer.MaxContextSizeToKeep,
+                HttpPooledConnectionLifetime = DocumentConventions.DefaultForServer.HttpPooledConnectionLifetime,
+                UseHttpCompression = _serverStore.Configuration.Sharding.ShardExecutorUseHttpCompression,
+                UseHttpDecompression = _serverStore.Configuration.Sharding.ShardExecutorUseHttpDecompression,
+                GlobalHttpClientTimeout = _serverStore.Configuration.Sharding.OrchestratorTimeoutInMinutes.AsTimeSpan,
+                CreateHttpClient = handler =>
+                {
+                    handler.ServerCertificateCustomValidationCallback = ShardingCustomValidationCallback;
```

Comment:
Instead of modifying ShardingCustomValidationCallback you could use just decide where you connect to.

https://pastebin.com/bj7vvca6

The client and server cert validation then stays as is and TLS layer is untouched.

Reply:
ping?